ios/engine/KMEI/KeymanEngine/Classes/Keyboard/InputViewController.swift

```diff
@@ -294,18 +294,65 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       self.swallowBackspaceTextChange = false
       return
     }
-    
+
     let contextBeforeInput = textDocumentProxy.documentContextBeforeInput ?? ""
     let selection = textDocumentProxy.selectedText ?? ""
     let contextAfterInput = textDocumentProxy.documentContextAfterInput ?? ""
     let context = "\(contextBeforeInput)\(selection)\(contextAfterInput)"
-    let bLength = contextBeforeInput.utf16.count
-    let sLength = selection.utf16.count
+    let bLength = contextBeforeInput.count
+    let sLength = selection.count
```

Comment:
What's the purpose of this change? What is it fixing? I note there is no documentation on what units `setContextState` is expecting (UTF-16 code units, USV codepoints, something else?), nor any documentation on KeymanWeb's `keyman.context.setSelection()` either -- which is what is ultimately called here.

From my analysis, it _looks_ like `keyman.context.setSelection()` should be taking USV -- definitely not Swift's 'character' count which is based on graphemes.

```suggestion
    let bLength = contextBeforeInput.unicodeScalars.count
    let sLength = selection.unicodeScalars.count
```

Reply:
Ooh, good catch with the `.unicodeScalars` bit.  That does seem to be behaving more in line with what we want.